src/jllvm/materialization/ByteCodeCompileLayer.cpp

```diff
@@ -1185,38 +1203,32 @@ void codeGenBody(llvm::Function* function, const Code& code, const ClassFile& cl
                 builder.CreateBr(basicBlocks[index]);
             },
             // TODO: GotoW
-            [&](I2B)
-            {
-                llvm::Value* value = operandStack.pop_back(builder.getInt32Ty());
-                llvm::Value* truncated = builder.CreateTrunc(value, builder.getInt8Ty());
-                operandStack.push_back(builder.CreateSExt(truncated, builder.getInt32Ty()));
-            },
-            [&](I2C)
-            {
-                llvm::Value* value = operandStack.pop_back(builder.getInt32Ty());
-                llvm::Value* truncated = builder.CreateTrunc(value, builder.getInt16Ty());
-                operandStack.push_back(builder.CreateZExt(truncated, builder.getInt32Ty()));
-            },
-            [&](I2D)
-            {
-                llvm::Value* value = operandStack.pop_back(builder.getInt32Ty());
-                operandStack.push_back(builder.CreateSIToFP(value, builder.getDoubleTy()));
-            },
-            [&](I2F)
-            {
-                llvm::Value* value = operandStack.pop_back(builder.getInt32Ty());
-                operandStack.push_back(builder.CreateSIToFP(value, builder.getFloatTy()));
-            },
-            [&](I2L)
-            {
-                llvm::Value* value = operandStack.pop_back(builder.getInt32Ty());
-                operandStack.push_back(builder.CreateSExt(value, builder.getInt64Ty()));
-            },
-            [&](I2S)
+            [&](OneOf<I2B, I2C, I2D, I2F, I2L, I2S>)
             {
                 llvm::Value* value = operandStack.pop_back(builder.getInt32Ty());
-                llvm::Value* truncated = builder.CreateTrunc(value, builder.getInt16Ty());
-                operandStack.push_back(builder.CreateSExt(truncated, builder.getInt32Ty()));
+
+                auto* result = match(
```

Comment:
I am somewhat indecisive on whether in this very case in particular, the one larger case with a nested match is nicer code due to how short the code before and after it is (so somewhat getting to that "Two time nesting for every op" aesthetic we previously discussed).

I'll let you be judge whether you think this is nicer or whether you'd rather leave it as is.

Reply:
Let's revert it.